controllers/vrg_volrep.go

```diff
@@ -318,9 +318,14 @@ func (v *VRGInstance) preparePVCForVRDeletion(pvc *corev1.PersistentVolumeClaim,
 	// For Sync mode, we don't want to set the retention policy to delete as
 	// both the primary and the secondary VRG map to the same volume. The only
 	// state where a delete retention policy is required for the sync mode is
-	// when the VRG is primary.
-	if !(v.instance.Spec.ReplicationState == ramendrv1alpha1.Secondary &&
-		v.instance.Spec.Sync.Mode == ramendrv1alpha1.SyncModeEnabled) {
+	// when the VRG is primary. Furthermore, we need to clear the PV claimRef
+	// in order for the PV to go to the Available status phase.
+	if v.instance.Spec.ReplicationState == ramendrv1alpha1.Secondary &&
+		v.instance.Spec.Sync.Mode == ramendrv1alpha1.SyncModeEnabled {
+		if err := v.clearPVClaimRef(pvc); err != nil {
```

Comment:
Technically the PV is now Available for another PVC request of the same size and StorageClass, the k8s provisioner hence may bind this PV to the new PVC request in another namespace and leak storage across namespaces.

Afaict, all Available PVs are looked at and if any satisfies the request it is bound to the new PVC request.

I do not have an alternative to suggest though, but we may want to track this caveat.

Reply:
That's true. We can put a dummy PVC name there as in Metro and in this PR, we don't check whether ClaimRef is nil or not.  We blindly assign to it the ClaimRef from the PV that we downloaded from the s3 store.  So if we stick in the ClaimRef of the existing PV a random name/namespace, like GUID, then that would solve the issue.

Reply:
What if we just remove the UUID and the resource version of the claimref. That way we ensure that the name of the pvc matches. I tried this on a minikube instance and it worked.

Reply:
@raghavendra-talur, I think that should work, but I was thinking something like this
```
pv.Spec.ClaimRef = &corev1.ObjectReference {
		Name: pv.Spec.ClaimRef.Name + pv.Spec.ClaimRef.ResourceVersion,
		Namespace: pv.Spec.ClaimRef.Namespace + pv.Spec.ClaimRef.ResourceVersion,
	}
```

Reply:
But if we only remove the UID and the Resource version, then it should work without the changes that I made for restore, does that sound right?

Reply:
Yes, no changes required during the restore.

Reply:
Ok, I am going to leave this PR open, we can close it later, but I will create a different PR with only the changes we talked about.